utils/mil_ros_tools/online_bagger/nodes/online_bagger.py

```diff
@@ -0,0 +1,427 @@
+#!/usr/bin/env python
+
+from __future__ import division
+import rospy
+import rosbag
+import rostopic
+import os
+import rospkg
+import time
+import math
+import resource
+from collections import deque
+import itertools
+import datetime
+from mil_ros_tools.srv import  BaggerCommands
+
+"""
+To call call service call the '/online_bagger/dump' service with the BaggerCommands.srv
+Amount and Unit. Doing this will generate a rosbag with the
+last x number of seconds saved
+
+For example:
+
+        bagging_server = rospy.ServiceProxy('/online_bagger/dump', BaggerCommands)
+        bag_status = bagging_server(x, 's')
+        bag_status = bagging_server(Amount, Unit)
+        amount = float32 (leave blank to dump entire bag)
+        unit = string (select units 's' or 'm')
+        """
+
+class OnlineBagger(object):
+
+    def __init__(self):
+
+        """
+        Make dictionary of dequeues.
+        Subscribe to set of topics defined by the yaml file in directory
+        Stream topics up to a given ram limit, dump oldest messages when limit is reached
+        Set up service to bag n seconds of data default to all of available data
+        """
+
+        self.i = 0  # successful subscriptions
```

Comment:
Might call this `successful_subscription_count` and `self.n` could be `iteration_count` or something more descriptive